marklogic-client-api/src/test/java/com/marklogic/client/test/rows/ValidateDocTest.java

```diff
@@ -86,10 +92,69 @@ public void xmlSchema() {
         //expectedUris.forEach(uri -> assertTrue("Did not find URI: " + uri, persistedUris.contains(uri)));
     }
+    @Test
+    public void testXmlSchemaWithStrictMode() {
+        if (!Common.markLogicIsVersion11OrHigher()) {
+            return;
+        }
+        String[][] triples = new String[][]{
+                new String[]{"http://example.org/rowgraph/s1", "http://example.org/rowgraph/p1", "http://example.org/rowgraph/o1"},
+                new String[]{"http://example.org/rowgraph/s1", "http://example.org/rowgraph/p2", "http://example.org/rowgraph/o2"},
+                new String[]{"http://example.org/rowgraph/s2", "http://example.org/rowgraph/p1", "http://example.org/rowgraph/o3"}
+        };
+        DatabaseClient client = Common.newClient();
+        XMLDocumentManager docMgr = client.newXMLDocumentManager();
+
+        String triplesXML =
+                "<sem:triples xmlns:sem=\"http://marklogic.com/semantics\">\n"+
+                        String.join("\n", (String[]) Arrays
+                                .stream(triples)
+                                .map(triple ->
+                                        "<sem:triple>"+
+                                                "<sem:subject>"+triple[0]+"</sem:subject>"+
+                                                "<sem:predicate>"+triple[1]+"</sem:predicate>"+
+                                                "<sem:object>"+triple[2]+"</sem:object>"+
+                                                "</sem:triple>"
+                                )
+                                .toArray(size -> new String[size]))+
+                        "</sem:triples>";
+        DocumentWriteSet documentWriteSet = docMgr.newWriteSet();
+        for(int i=0; i<4; i++){
+            documentWriteSet.add("/acme/"+i+".xml", new StringHandle(triplesXML).withFormat(Format.TEXT));
+        }
+        docMgr.write(documentWriteSet);
+
+        PlanBuilder.Plan plan = op
+                .fromDocUris(op.cts.directoryQuery("/acme/"))
+                .joinDoc(op.col("doc"), op.col("uri"))
+                .validateDoc(op.col("doc"),
+                        op.schemaDefinition("xmlSchema").withMode("strict")
+                );
+        Iterator<RowRecord> rows = rowManager.resultRows(plan).iterator();
+        Set<String> uris = new HashSet<>();
+        while (rows.hasNext()){
+            RowRecord rowRecord = rows.next();
+            if(rowRecord.getString("uri") != null)
+                uris.add(rowRecord.getString("uri"));
+        }
+        // TODO : uncomment below after https://bugtrack.marklogic.com/57987 is fixed
+        // assertTrue(uris.size() == 4);
+        // assertTrue(uris.contains("/acme/1.xml"));
+        //assertTrue(uris.contains("/acme/2.xml"));
+        //assertTrue(uris.contains("/acme/3.xml"));
+        //assertTrue(uris.contains("/acme/0.xml"));
+
+        QueryManager queryManager = client.newQueryManager();
```

Comment:
No need for these lines of code, the parent class will delete data at the start of each test.

Reply:
Actually, these docs were not removed after running all the tests. Was it because a different user was used? (line 105). Also, i had to use the rest-writer because the user test-rest-writer was giving exception while inserting docs, something like  - "Atleast one update permission needed while inserting documents"

Reply:
That's a good exception! We don't want to rely on users having default permissions because it can hide actual problems. Just use "newDefaultMetadata()" (which is in AbstractOpticUpdateTest) to construct a `DocumentMetadataHandle`, it has read/update permissions on it. 

Also, the docs should not be removed after a test completes - that makes debugging failed tests much easier. Instead, docs should be removed at the start of a test, which the parent class does by deleting everything in the "/acme/" directory.